pkg/controllers/machine/disruption/drift.go

```diff
@@ -28,70 +28,72 @@ import (
 
 	"github.com/aws/karpenter-core/pkg/apis/settings"
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/apis/v1beta1"
 	"github.com/aws/karpenter-core/pkg/cloudprovider"
 	"github.com/aws/karpenter-core/pkg/scheduling"
+	machineutil "github.com/aws/karpenter-core/pkg/utils/machine"
 )
 
-// Drift is a machine sub-controller that adds or removes status conditions on drifted machines
+// Drift is a nodeClaim sub-controller that adds or removes status conditions on drifted nodeClaims
 type Drift struct {
 	cloudProvider cloudprovider.CloudProvider
 }
 
-func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (reconcile.Result, error) {
-	hasDriftedCondition := machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted) != nil
+func (d *Drift) Reconcile(ctx context.Context, nodePool *v1beta1.NodePool, nodeClaim *v1beta1.NodeClaim) (reconcile.Result, error) {
+	hasDriftedCondition := nodeClaim.StatusConditions().GetCondition(v1beta1.NodeDrifted) != nil
 
 	// From here there are three scenarios to handle:
-	// 1. If drift is not enabled but the machine is drifted, remove the status condition
+	// 1. If drift is not enabled but the NodeClaim is drifted, remove the status condition
 	if !settings.FromContext(ctx).DriftEnabled {
-		_ = machine.StatusConditions().ClearCondition(v1alpha5.MachineDrifted)
+		_ = nodeClaim.StatusConditions().ClearCondition(v1beta1.NodeDrifted)
 		if hasDriftedCondition {
-			logging.FromContext(ctx).Debugf("removing drift status condition from machine as drift has been disabled")
+			logging.FromContext(ctx).Debugf("removing drift status condition, drift has been disabled")
 		}
 		return reconcile.Result{}, nil
 	}
-	// 2. If Machine is not launched, remove the drift status condition
-	if launchCond := machine.StatusConditions().GetCondition(v1alpha5.MachineLaunched); launchCond == nil || launchCond.IsFalse() {
-		_ = machine.StatusConditions().ClearCondition(v1alpha5.MachineDrifted)
+	// 2. If NodeClaim is not launched, remove the drift status condition
+	if launchCond := nodeClaim.StatusConditions().GetCondition(v1beta1.NodeLaunched); launchCond == nil || launchCond.IsFalse() {
+		_ = nodeClaim.StatusConditions().ClearCondition(v1beta1.NodeDrifted)
 		if hasDriftedCondition {
-			logging.FromContext(ctx).Debugf("removing drift status condition from machine as machine isn't launched")
+			logging.FromContext(ctx).Debugf("removing drift status condition, isn't launched")
```

Comment:
Can you take a note to look at this one? Seems like this case should never happen, and is potentially just defensive programming